libkbfs/keybase_daemon_rpc.go

```diff
@@ -305,3 +306,46 @@ func (k *KeybaseDaemonRPC) Shutdown() {
 		k.shutdownFn()
 	}
 }
+
+// GetTLFCryptKeys implements the TlfKeysInterface interface for
+// KeybaseDaemonRPC.
+func (k *KeybaseDaemonRPC) GetTLFCryptKeys(
+	ctx context.Context, tlfName string) (res keybase1.TLFCryptKeys, err error) {
+	var tlfHandle *TlfHandle
+
+getHandle:
+	for {
+		tlfHandle, err = ParseTlfHandle(ctx, k.config.KBPKI(), tlfName, false)
+		switch e := err.(type) {
+		case TlfNameNotCanonical:
+			tlfName = e.NameToTry
+		case nil:
+			break getHandle
+		default:
+			return res, err
+		}
+	}
+
+	res.CanonicalName = keybase1.CanonicalTlfName(tlfHandle.GetCanonicalName())
+
+	var rmd ImmutableRootMetadata
+	var id TlfID
+	_, rmd, id, err = k.config.KBFSOps().GetOrInitializeNewMDMaster(
+		ctx, k.config.MDOps(), tlfHandle)
+	if err != nil {
+		return res, err
+	}
+	res.TlfID = keybase1.TLFID(id.String())
+
+	var keys []TLFCryptKey
+	keys, err = k.config.KeyManager().GetTLFCryptKeyFromAllGenerations(ctx, rmd)
+	if err != nil {
+		return res, err
+	}
+
+	res.FirstValidKeyGen = FirstValidKeyGen
```

Comment:
I thought about using a slice of `{generation, key}` struct, but decided to use to save a few bytes. It's probably not worth it. Are you OK with using struct slice here (so that keys are sorted by generation) as well?